sympy/physics/vector/tests/test_point.py

```diff
@@ -126,3 +128,44 @@ def test_point_partial_velocity():
     assert p.partial_velocity(N, u1) == A.x
     assert p.partial_velocity(N, u1, u2) == (A.x, N.y)
     raises(ValueError, lambda: p.partial_velocity(A, u1))
+
+def test_point_vel():
+    q1, q2 = dynamicsymbols('q1 q2')
+    N = ReferenceFrame('N')
+    B = ReferenceFrame('B')
+    O = Point('O')
+    Q = Point('Q')
+    Q.set_pos(O, q1 * N.x)
+    raises(ValueError , lambda : Q.vel(N))
+    O.set_vel(N, q2 * N.y)
+    assert Q.vel(N) == Derivative(q1) * N.x + q2 * N.y
+    P1 = Point('P1')
+    P1.set_pos(O, q1 * B.x)
+    raises(ValueError, lambda : P1.vel(B))
+    raises(ValueError, lambda : P1.vel(N))
+    P2 = Point('P2')
+    P2.set_pos(P1, q2 * B.z)
+    raises(ValueError, lambda : P2.vel(B))
+    P3 = Point('P3')
+    P3.set_pos(P2, q1 * B.x)
+    raises(ValueError, lambda : P3.vel(B))
+    P5 = Point('P5')
+    P5.set_pos(P2, q1 * N.x)
+    raises(ValueError, lambda : P5.vel(N))
+    P1.set_vel(B, 10 * q1 * B.x)
+    assert P3.vel(B) == (10 * q1 + Derivative(q1)) * B.x + Derivative(q2) * B.z
+    P4 = Point('P4')
+    P4.set_pos(O, q1 * B.x)
+    raises(ValueError, lambda : P4.vel(N))
+    Q1 = Point('Q1')
+    Q1.set_vel(N, q1 * N.x)
+    Q2 = Point('Q2')
+    Q3 = Point('Q3')
+    Q4 = Point('Q4')
+    Q2.set_pos(Q1, q2 * N.x)
+    Q3.set_pos(Q2, q1 * N.x)
+    Q4.set_pos(Q3, q2 * N.x)
+    assert Q4.vel(N) == (q1 + Derivative(q1) + 2 * Derivative(q2)) * N.x
+    Q3.set_vel(N, q1 * N.x)
+    assert Q3.vel(N) == q1 * N.x
+    assert Q4.vel(N) == (q1 + Derivative(q2)) * N.x
```

Comment:
Please add some comments to the file explaining what your tests do.

Reply:
> Please add some comments to the file explaining what your tests do.

Sure sir